File: vaultx-frontend/src/services/shareService.js
import api from './api';

const shareService = {
  createShare: (data) => api.post('/shares', data),
  getUserShares: (page = 0, size = 10) => api.get(`/shares?page=${page}&size=${size}`),
  getShare: (id) => api.get(`/shares/${id}`),
  deleteShare: (id) => api.delete(`/shares/${id}`),
  revokeShare: (id) => api.post(`/shares/${id}/revoke`),
  updatePassword: (id, password) => api.post(`/shares/${id}/password`, null, { params: { password } })
};

export default shareService;

File: vaultx-backend/src/main/java/com/vaultx/service/ShareService.java
package com.vaultx.service;

import com.vaultx.common.PagedResponse;
import com.vaultx.dto.share.ShareCreateRequest;
import com.vaultx.dto.share.ShareDto;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface ShareService {
    ShareDto createShare(ShareCreateRequest request);
    PagedResponse<ShareDto> getUserShares(Pageable pageable);
    ShareDto getShare(UUID id);
    void revokeShare(UUID id);
    void deleteShare(UUID id);
    void updatePassword(UUID id, String newPassword);
}
